.github/workflows/end2end.yaml

```diff
@@ -0,0 +1,406 @@
+  AWS_CRR_BUCKET_NAME: ci-zenko-aws-crr-target-bucket
+  AZURE_CRR_BUCKET_NAME: ci-zenko-azure-crr-target-bucket
+  GCP_CRR_BUCKET_NAME: ci-zenko-gcp-crr-target-bucket
+  GCP_CRR_MPU_BUCKET_NAME: ci-zenko-gcp-crr-mpu-bucket
+  GCP_ACCESS_KEY: ${{ secrets.AWS_GCP_BACKEND_ACCESS_KEY }}
+  GCP_SECRET_KEY: ${{ secrets.AWS_GCP_BACKEND_SECRET_KEY }}
+  GCP_BACKEND_SERVICE_KEY: ${{ secrets.GCP2_SERVICE_KEY }}
+  GCP_BACKEND_SERVICE_EMAIL: ${{ secrets.GCP2_SERVICE_EMAIL }}
+  # Enable this for Ring tests
+  ENABLE_RING_TESTS: "false"
+  RING_S3C_ACCESS_KEY: ${{ secrets.RING_S3C_BACKEND_ACCESS_KEY }}
+  RING_S3C_SECRET_KEY: ${{ secrets.RING_S3C_BACKEND_SECRET_KEY }}
+  RING_S3C_ENDPOINT: ${{ secrets.RING_S3C_BACKEND_ENDPOINT }}
+  RING_S3C_BACKEND_SOURCE_LOCATION: rings3cbackendingestion
+  RING_S3C_INGESTION_SRC_BUCKET_NAME: ingestion-test-src-bucket-${{ github.run_id }}-${{ github.run_attempt }}
+  SKOPEO_PATH: "/tmp"
+  HELM_VERSION: "v3.5.3"
+  YQ_VERSION: "v4.6.3"
+  YQ_BINARY: "yq_linux_amd64"
+  KUSTOMIZE_VERSION: "v4.4.1"
+  GO_VERSION: "1.16.2"
+  SKOPEO_VERSION: "v1.5.2"
+  KUBECTL_VERSION: "1.21.1"
+  TILT_VERSION: "0.23.4"
+  KIND_VERSION: "v0.11.1"
+
+jobs:
+  check-dashboard-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: install jq
+        run: |
+          sudo apt-get update && sudo apt-get install -y \
+          jq
+      - name: Verify monitoring dashboard versions
+        run: bash ./.github/scripts/check_versions.sh
+
+  build-doc:
+    runs-on: ubuntu-latest
+    env:
+      ARTIFACTS_USER: "${{ secrets.ARTIFACTS_USER }}"
+      ARTIFACTS_PASSWORD: "${{ secrets.ARTIFACTS_PASSWORD }}"
+      DEBIAN_FRONTEND: "noninteractive"
+      ROOT_DOCUMENTATION_BUILDDIR: "docs/docsource/build"
+      DOCUMENTATION_BUILDDIR: "build"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install required dependencies
+        uses: ./.github/actions/install-doc-dependencies
+      - name: Create build directory
+        run: mkdir ./${{ env.DOCUMENTATION_BUILDDIR }}
+        working-directory: ./docs
+      - name: Build docs
+        run: tox --workdir /tmp/tox -e docs -- html BUILDDIR=${{ env.DOCUMENTATION_BUILDDIR }}
+        working-directory: ./docs
+      - name: Create tar file for documentation
+        run: |
+          tar --create --gzip --directory ./${{ env.ROOT_DOCUMENTATION_BUILDDIR }} --file ./${{ env.ROOT_DOCUMENTATION_BUILDDIR }}/zenko-documentation.tar.gz doctrees html
+      - name: Upload documentation
+        uses: scality/action-artifacts@v3
+        with:
+          method: upload
+          url: https://artifacts.scality.net
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+          source: ${{ env.ROOT_DOCUMENTATION_BUILDDIR }}
+
+  build-iso:
+    runs-on:
+      - self-hosted
+      - centos7
+      - large
```

Comment:
Do you need centos7 here?

I would recommend building the iso with a docker image. That will allow you to use it locally to build the iso as well when required.

You can use then the `container` parameter in a job to use that image inside the pipeline.

Reply:
I am not sure if this can help, considering the changes required to be able to send the generated iso file from a docker container to the artifacts... It will require a bunch of secrets to be set locally if we want to use it as well as ability to write the file in the host system, with the support of both the ISO and Base-ISO files.

So I won't do this in this PR as this requires proper end2end testing and I do not have the network bandwidth to test that locally currently.

For the Centos7 part, I guess we can get rid of it and use ubuntu, but I do not see any benefit of doing so...

Reply:
Feel free to schedule a session with me some time next week. I think you're missing some GitHub Actions knowledge here. There won't be any change to upload the iso to artifacts, and managing secrets will remain the same. Checkout https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container for more information

Reply:
Just to confirm: do you mean having a dockerfile + an entrypoint where everything is done (so that we can just run it locally and call it in the CI with 1 command), or just running the steps not with `ubuntu-latest` but inside a container where all the depds are already installed?

Reply:
I'll try to schedule something next week then, thanks!